pkgs/development/python-modules/sopel/default.nix

```
@@ -52,6 +52,8 @@ buildPythonPackage rec {
     popd
   '';
 
+  pythonImportsCheck = [ "sopel" ];
```

@jonringer Is it worth it adding an imports check, when we do have the testsuite?

---

One reason to keep it might be that if a future version bump moves the tests to a different directory or something in such a way that pytestCheckHook doesn't see them, the build could pass without the tests having _actually_ ran.

---

I'd really wish the pytestCheckHook would fail hard in such a case.

---

That does sound better. It's on the TODO list to add a check to hammering/@r-rmcgibbo for "pytest ran zero tests", but failing inside `pytestCheckHook` is obviously much more robust than a "suggestion" from a bot.

---

From my point of view, adding `pythonImportsCheck` is superfluous if the tests are ran but it's a pretty "cheap" tests and an additional safeguard.

I will continuing to add `pythonImportsCheck` for now if I touch a package.

---

Yes, it is. It makes sure the package can be imported at all.

---

It's worth adding because sometimes a package will rely on `setuptools` to be present, and that might be available during the build, but not at runtime.